Library/PushNotificationdialogTests.swift

```diff
@@ -81,4 +81,18 @@ class PushNotificationDialogTests: XCTestCase {
       XCTAssertEqual(["login", "save"], userDefaults.deniedNotificationContexts )
     }
   }
+
+  func testLoginContextResets_AfterLogout() {
+
+    withEnvironment(userDefaults: userDefaults) {
+
+      PushNotificationDialog.didDenyAccess(for: .login)
+      PushNotificationDialog.didDenyAccess(for: .save)
```

Comment:
Shouldn't we also reset the `.save` context? Since "saving" is a logged-in action.

Reply:
The prompt should appear within the `.save` context only for the first ever project saved. The initial requirement was to show the Contextual Dialog up to 3x to avoid overwhelming our users with alerts popping up on the screen. But this is a good question. I will check w/ @dannyalright and see what would be the best UX.

Reply:
yeah let's chat in-person @Scollaco (and @ifbarrera if you'd like to join)